sunpy/coordinates/sun.py

```diff
@@ -100,13 +100,30 @@ def carrington_rotation_time(crot):
 
     Parameters
     ----------
-    crot : float
-        Carrington rotation number. Can be a fractional cycle number.
+    crot : int, float, numpy.array
+        Carrington rotation number. Can be a integral or fractional cycle number
+        or a Numpy array of such Carrington rotation numbers. If a single
+        Carrington rotation number is passed with multiple logitudes then
+        rotation times at each of those longitudes is calculated.
+
+    longitude : astropy.units.Quantity
+        Carrington longitude (in degrees).If a single longitude is passed with
+        multiple Carrington rotation numbers then the same rotation is applied
+        to each Carrington rotation number while calculating rotation times.
+        When this argument is passed Carrington rotation number(s) should be
+        strictly integral.
 
     Returns
     -------
     astropy.time.Time
     """
+    if longitude is not None:
+        crot = np.array(crot)
+        longitude = np.array(longitude.to('deg').value)
+        assert not (crot % 1).all(), "Carrington Number(s) should be integral"
+        assert (longitude > 0).all() and (longitude <= 360).all(), \
```

Comment:
These should not be `assert` checks, but rather `quantity_allclose()` checks that raise a `ValueError` when needed

Reply:
It's kinda hard to formulate the second condition into an `quantity_allclose()` so I let it be. Did you have some kind of formulation in mind for that?

Reply:
No, you're right: that doesn't need to use `quantity_allclose()`